archaius2-guice/src/main/java/com/netflix/archaius/guice/ArchaiusModule.java

```diff
@@ -150,169 +112,143 @@ public ArchaiusModule withCascadeStrategy(Class<? extends CascadeStrategy> casca
         return this;
     }
     
-    private String getUniqueName(String prefix) {
-        uniqueNameCounter++;
-        return prefix +"-" + uniqueNameCounter;
+    protected void configureArchaius() {
```

Comment:
I thought there was a recommendation to discourage sub-classes and if possible make the module final. In particular, how does the deduping still work if a module is registered along with a subclass of that module?